components/calendar/test/calendar.visual-diff.js

```diff
@@ -0,0 +1,302 @@
+const puppeteer = require('puppeteer');
+const VisualDiff = require('@brightspace-ui/visual-diff');
+
+describe('d2l-calendar', () => {
+
+	const visualDiff = new VisualDiff('calendar', __dirname);
+
+	let browser, page;
+
+	const firstCalendarOfPage = '#contains-today-diff-selected';
+
+	before(async() => {
+		browser = await puppeteer.launch();
+		page = await browser.newPage();
+		await page.setViewport({width: 400, height: 1600, deviceScaleFactor: 2});
+		await page.goto(`${visualDiff.getBaseUrl()}/components/calendar/test/calendar.visual-diff.html`, {waitUntil: ['networkidle0', 'load']});
+		await page.bringToFront();
+	});
+
+	after(() => browser.close());
+
+	it('has no selected-value when selected-value not set', async function() {
+		const rect = await visualDiff.getRect(page, '#no-selected');
+		await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+	});
+
+	it('month with first row entirely current month days and last row containing next month days', async function() {
+		const rect = await visualDiff.getRect(page, '#dec-2019');
+		await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+	});
+
+	describe('localization', () => {
+
+		after(async() => {
+			await page.evaluate(() => document.querySelector('html').setAttribute('lang', 'en'));
+		});
+
+		[
```

Comment:
What's the rationale for screenshots of all the languages? Is this already covered by localization tests?

Reply:
Mostly that the calendar changes dramatically based on the language -- day headers, first-day-of-the-week most notably. Normally we wouldn't care about multiple languages, but I feel like in calendar's case it's important.